spectrocloud/resource_cluster_eks.go

```diff
@@ -302,9 +302,10 @@ func resourceClusterEks() *schema.Resource {
 							Required: true,
 						},
 						"capacity_type": {
-							Type:     schema.TypeString,
-							Default:  "on-demand",
-							Optional: true,
+							Type:         schema.TypeString,
+							Default:      "on-demand",
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"on-demand", "spot"}, false),
```

Comment:
here same as above. I suggest to actually move common parts of schema into a separate file. Could you please check how it's done in the example schemas/location_cluster_config.go ?